packages/@react-spectrum/table/src/TreeGridTableView.tsx

```diff
@@ -1387,16 +1390,21 @@ function TableCheckboxCell({cell}) {
   );
 }
 
-function TableCell({cell}) {
-  let {state} = useTableContext();
+function TableCell({cell, density, scale}) {
+  let {state, toggleKey, direction} = useTableContext();
   let ref = useRef();
   let columnProps = cell.column.props as SpectrumColumnProps<unknown>;
   let isDisabled = state.disabledKeys.has(cell.parentKey);
   let {gridCellProps} = useTableCell({
     node: cell,
     isVirtualized: true
   }, state, ref);
-
+  let showExpandCollapseButton = cell.index === 0 && (state.collection.getItem(cell.parentKey)?.props.childItems?.length > 0 || state.collection.getItem(cell.parentKey)?.props.children.length > state.collection.columnCount);
+  let isExpanded = showExpandCollapseButton && (state.expandedKeys === 'all' || state.expandedKeys.has(cell.parentKey));
+  let levelOffset = (cell.level - (showExpandCollapseButton ? 1 : 2)) * 16 + 4;
```

Comment:
I'll test it out, but are the numbers included in here and other places in the PR accounting for scale changes as well?

Reply:
I can fix this, and also add a comment explaining the logic